src/common/contexts/provider.tsx

```diff
@@ -99,9 +101,25 @@ export const ProviderContextProvider: FunctionComponent<ProviderContextProviderP
   );
   const [provider, setProvider] = useState<providers.Provider | undefined>(defaultProvider.current);
 
+  const { showOverlay, setOverlayVisible } = useContext(OverlayContext);
+  const closeOverlay = () => {
+    showOverlay(undefined);
+    setOverlayVisible(false);
+  };
+
   const changeNetwork = async (chainId: ChainId) => {
-    await walletSwitchChain(chainId);
-    setCurrentChainId(chainId);
+    try {
+      showOverlay(<LoadingModal title={"Changing Network..."} content={"Please respond to the metamask window"} />);
+      await walletSwitchChain(chainId);
+      setCurrentChainId(chainId);
+      closeOverlay();
+    } catch (e: any) {
+      showOverlay(
+        showDocumentTransferMessage("You've cancelled changing network.", {
+          isSuccess: false,
+        })
+      );
+    }
```

Comment:
dont couple change network with displaying the modal, do it in the auto-switching network context. its fine to duplicate the code there.

Reply:
ok decoupled and created new network context to handle the displaying of modal.